ui-admin/src/forms/designer/questions/VisibilityFields.tsx

```diff
@@ -37,44 +39,47 @@ export const VisibilityFields = (props: VisibilityFieldsProps) => {
   )
 
   return (
-    <>
-      <div className="mb-3">
-        <Checkbox
-          checked={hasVisibleIfExpression}
-          disabled={disabled}
-          label="Conditionally show this question"
-          onChange={checked => {
-            if (checked) {
-              onChange({
-                ...question,
-                visibleIf: ''
-              })
-            } else {
-              onChange(unset('visibleIf', question))
-            }
-          }}
-        />
-      </div>
-
-      {hasVisibleIfExpression && (
+    <div className="bg-white rounded-3 p-2 mb-2">
+      <CollapsibleSectionButton targetSelector={`#${targetId}`} sectionLabel={'Visibility'}/>
+      <div className="collapse hide" id={targetId}>
         <div className="mb-3">
-          <TextInput
-            infoContent={infoPopupContent}
-            // eslint-disable-next-line max-len
-            description={'If this expression evaluates to true, the question will be shown.'}
+          <Checkbox
+            checked={hasVisibleIfExpression}
             disabled={disabled}
-            label="Visibility expression"
-            required={true}
-            value={question.visibleIf}
-            onChange={value => {
-              onChange({
-                ...question,
-                visibleIf: value
-              })
+            label="Conditionally show this question"
+            onChange={checked => {
+              if (checked) {
+                onChange({
+                  ...question,
+                  visibleIf: ''
+                })
+              } else {
+                onChange(unset('visibleIf', question))
+              }
             }}
           />
         </div>
-      )}
-    </>
+
+        {hasVisibleIfExpression && (
+          <div className="mb-3">
+            <TextInput
+              infoContent={infoPopupContent}
+              // eslint-disable-next-line max-len
+              description={'If this expression evaluates to true, the question will be shown.'}
+              disabled={disabled}
+              label="Visibility expression"
+              required={true}
+              value={question.visibleIf}
+              onChange={value => {
+                onChange({
+                  ...question,
+                  visibleIf: value
+                })
+              }}
+            />
+          </div>
```

Comment:
I noticed that when I clicked `Conditionally show this question` the field shows up immediately being red but don't show up at the top as an error that blocks saving.

I would expect that it would show up red only if I put stuff in it and cleared it, or tried to save it without providing a value. I noticed this was a pattern across multiple flows - if I add a new question for example, a bunch of stuff shows up immediately red.

I don't mind the way it is now, but I wonder if it would be good to beef up our client side validation so that missing visibility expressions show up as errors that block saving, and none of these fields are red by default, only upon saving. Just something to consider 😄

Reply:
Agreed- the validation is super rough right now. I've added this to the list of "functionality still to come" in the PR description, which we can turn into backlog tickets.